### PR TITLE
Add Email free trial pricing to email comparison page

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* This PR adds a new `titan/free-trial` feature flag and adds logic to the email comparison page to show the correct pricing for each domain based on the eligibility data we get from the server.

#### Screenshot

Three month trial

#### Testing instructions

This is going to need coordination with the back end to work, and requires that you can enable store sandbox. (See PCYsg-IA-p2 for options, depending on whether you have a developer back end.)

The next steps depend on whether D59149-code has been merged or not.

##### The change above has _not_ been merged

* Enable store sandbox, and ensure that your currency is not set to USD, DKK, or EUR.
  - If you have a developer back end, you can take a look at the diff above to support any of these currencies
* Navigate to the [live branch](https://calypso.live/?branch=add/titan-free-trial-pricing), and once it has rendered, add `?flags=titan/free-trial` to the URL.
* Try to purchase Email for a domain where you haven't purchased Email before (this can be a new domain)
* Verify that the email comparison page shows you a three month discount for Email
* Add a mailbox, and verify that checkout also shows you a free purchase of a three month trial

##### The change above _has_ been merged

This will require a development back end, as the current rules require the domain to be purchased on or after 2021-03-29. (We may shift this date for store sandbox mode.)

* Enable store sandbox
* Navigate to the [live branch](https://calypso.live/?branch=add/titan-free-trial-pricing), and once it has rendered, add `?flags=titan/free-trial` to the URL.
* Try to purchase Email for a domain where you haven't purchased Email before (this can be a new domain)
* Verify that the email comparison page shows you a three month discount for Email
* Add a mailbox, and verify that checkout also shows you a free purchase of a three month trial

### CODE REVIEW

**Comment by daledupreez** [comment]
cc @saygunnyc for design review, though this one is _really_ small.

**Review by leonardost — APPROVED** [review]
LGTM! 👍

File: config/development.json [hunk 1]
```diff
@@ -179,6 +179,7 @@
 		"site-indicator": true,
 		"memberships": true,
 		"support-user": true,
+		"titan/free-trial": true,
```
Comment (hambai) [inline config/development.json]: Does this need a feature flag at this point?

**Review by hambai — APPROVED** [review]
Tested this and it works. I only wonder why we need the feature flag considering that the free trial is technically already enabled?

File: config/development.json (hunk 1 above)
Comment (daledupreez) [reply config/development.json]: Good point! I was originally hoping to merge this on Thursday or Friday, at which point that made sense.
I've removed the flag and the check.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5628945

Thank you @daledupreez for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5628945).

**Pull request merged**